src/Form/matching.js

```diff
@@ -160,17 +160,19 @@ class Matching {
      * @param {HTMLInputElement|HTMLSelectElement} input
      * @param {HTMLFormElement} formEl
      * @param {{isLogin?: boolean}} [opts]
-     * @returns {SupportedSubTypes | string}
+     * @returns {SupportedTypes}
      */
     inferInputType (input, formEl, opts = {}) {
         const presetType = input.getAttribute(ATTR_INPUT_TYPE)
+        // @ts-ignore
         if (presetType) return presetType
 
         // For CC forms we run aggressive matches, so we want to make sure we only
         // run them on actual CC forms to avoid false positives and expensive loops
         if (this.isCCForm(formEl)) {
             const ccMatchers = this.matcherList('cc')
             const subtype = this.subtypeFromMatchers(ccMatchers, input, formEl)
+            // @ts-ignore
             if (subtype) return `creditCard.${subtype}`
```

Comment:
This fails because "creditCard.emailAddress" isn't a thing for example. Perhaps there's a simpler way to narrow only to cc types?

Same for the other ignores.

Reply:
Again, maybe type casting can help?